File: SeoulTools/Data/Config/UnitTests/SlimCS/ScriptsCS/Engine/Core/Vector2D.cs
// Utility class, wraps a 2D vector.
//
// Copyright (c) Demiurge Studios, Inc.
// 
// This source code is licensed under the MIT license.
// Full license details can be found in the LICENSE file
// in the root directory of this source tree.

using System.Diagnostics.Contracts;

using static SlimCS;

public class Vector2D
{
	public Vector2D(double fX, double fY)
	{
		x = fX;
		y = fY;
	}

	[Pure]
	public double GetLength()
	{
		return math.sqrt(x * x + y * y);
	}

	[Pure]
	public static Vector2D operator+(Vector2D v1, Vector2D v2)
	{
		return new Vector2D(v1.x + v2.x, v1.y + v2.y);
	}

	[Pure]
	public static Vector2D operator-(Vector2D v1, Vector2D v2)
	{
		return new Vector2D(v1.x - v2.x, v1.y - v2.y);
	}

	[Pure]
	public static Vector2D operator*(Vector2D v, double fScalar)
	{
		return new Vector2D(v.x * fScalar, v.y * fScalar);
	}

	[Pure]
	public static Vector2D operator/(Vector2D v, double fScalar)
	{
		return new Vector2D(v.x / fScalar, v.y / fScalar);
	}

	[Pure]
	public override string ToString()
	{
		return $"{{x={x}, y={y}}}";
	}

	public double x;
	public double y;
}

File: SeoulTools/Data/Config/UnitTests/SlimCS/ScriptsDebug/Engine/Core/Vector2D.lua
-- Utility class, wraps a 2D vector.
--
-- Copyright (c) Demiurge Studios, Inc.
-- 
-- This source code is licensed under the MIT license.
-- Full license details can be found in the LICENSE file
-- in the root directory of this source tree.





local Vector2D = class 'Vector2D'

function Vector2D:Constructor(fX, fY)

	self.x = fX
	self.y = fY
end

function Vector2D:GetLength()


	return math.sqrt(self.x * self.x + self.y * self.y)
end

function Vector2D.__add(
	v1, v2)

	return Vector2D:New(v1.x + v2.x, v1.y + v2.y)
end

function Vector2D.__sub(
	v1, v2)

	return Vector2D:New(v1.x - v2.x, v1.y - v2.y)
end

function Vector2D.__mul(
	v, fScalar)

	return Vector2D:New(v.x * fScalar, v.y * fScalar)
end

function Vector2D.__div(
	v, fScalar)

	return Vector2D:New(v.x / fScalar, v.y / fScalar)
end

function Vector2D:__tostring()


	return '{x=' .. self.x .. ', y=' .. self.y .. '}'
end



return Vector2D
